src/deterministic_brb.rs

```diff
@@ -212,6 +224,32 @@ impl<A: Actor<S>, SA: SigningActor<A, S>, S: Sig, BRBDT: BRBDataType<A>>
         self.send(peer, payload)
     }
 
+    /// Resend any proof of agreements that we have not yet received delivery confirmation for.
+    #[allow(clippy::type_complexity)]
+    pub fn resend_pending_deliveries(
+        &self,
+    ) -> Result<Vec<Packet<A, S, BRBDT::Op>>, Error<A, S, BRBDT::ValidationError>> {
+        let mut packets = Vec::new();
+        for (msg, delivered) in self.pending_delivery.iter() {
+            let proof = if let Some(proof) = self.pending_proof.get(&msg).cloned() {
+                proof
+            } else {
+                continue;
+            };
+
+            let recipients = &self.membership.members(msg.gen)? - delivered;
+
+            packets.extend(self.broadcast(
+                &Payload::BRB(Op::ProofOfAgreement {
+                    msg: msg.clone(),
+                    proof,
+                }),
+                recipients,
+            )?)
+        }
+        Ok(packets)
+    }
+
     /// Initiates an operation for the BRBDataType being secured by BRB.
     #[allow(clippy::type_complexity)]
     pub fn exec_op(
```

Comment:
So I was thinking that exec_op would return some some kind of NotReadyForNextOpYet error if we did not yet have supermajority of ProofOfAgreement.   Also we could have exec_op_blocking() with optional timeout that would wait/block until we receive such.

I don't see that exec_op returns such an error, so did I miss it somehow, or else what is the feedback mechanism for the app to know to retry/wait?

Reply:
let's put an example usage into doc comment for exec_op.  along the lines of:

```
while !resend_pending_deliveries().is_empty() {
   sleep();
}
exec_op()?;
```